helix-core/src/test/java/org/apache/helix/integration/task/TestForceDeleteWorkflow.java

```diff
@@ -0,0 +1,337 @@
+package org.apache.helix.integration.task;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.helix.AccessOption;
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.HelixManager;
+import org.apache.helix.TestHelper;
+import org.apache.helix.task.JobConfig;
+import org.apache.helix.task.TaskConfig;
+import org.apache.helix.task.TaskConstants;
+import org.apache.helix.task.TaskState;
+import org.apache.helix.task.TaskUtil;
+import org.apache.helix.task.Workflow;
+import org.apache.helix.task.WorkflowContext;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/*
+This test checks the functionality of the forcedelete in various conditions.
+ */
+
+public class TestForceDeleteWorkflow extends TaskTestBase {
+  private HelixAdmin admin;
+  // Delay considered for force deletion and delay considered for disableling the  cluster.
+  private static final int DELETE_DELAY = 2000;
+  private long Disable_Delay = 1000L;
+
+  @BeforeClass
+  public void beforeClass() throws Exception {
+    super.beforeClass();
+    admin = _gSetupTool.getClusterManagementTool();
+    createManagers();
+
+  }
+
+  @Test
+  public void testDeleteCompletedWorkflowForcefully() throws Exception {
+    // Create a simple workflow and wait for its completion. Then delete the IdealState,
+    // WorkflowContext and WorkflowConfig.
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = new Workflow.Builder(workflowName);
+
+    //Workflow Schematic:
```

Comment:
I see a lot of duplicate code here for creating a workflow. Could you please refactor them?